libass/ass_drawing.c

```diff
@@ -43,93 +43,198 @@ static bool token_check_values(ASS_DrawingToken *token, int i, ASS_TokenType typ
     return true;
 }
 
+static inline void add_node(ASS_DrawingToken **tail, ASS_TokenType type, ASS_Vector point)
+{
+    assert(tail && *tail);
+
+    (*tail)->next = calloc(1, sizeof(**tail));
```

Comment:
I think reverse order is better, i. e.
```c
ASS_DrawingToken *new_tail = malloc(sizeof(**tail));
(*tail)->next = new_tail;
```
It should be easier to understand for compilers too, as it has less memory accesses.

As for `calloc` vs. `malloc`, I think that such "protective" programming is harmful: it would be easier to find a missing initialization (with valgrind, for example) in case of uninitialized value than if it was already zero-initialized. Also it's impossible to say by looking at the code if such zero initialization is intended or accidental and programmer forgot to set a correct value.

Reply:
Adopted the reversed order